lib/puppet/functions/ensure_packages.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# @summary Takes a list of packages and only installs them if they don't already exist.
```

Comment:
Technically you can also ensure they're absent via `default_attributes`, right?

Reply:
yes, or by specifying a hash for the first parameter.  But... I really don't know why people do that.  IMO it becomes a fairly  useless function the moment you start specifying attributes as they likely won't match with any other definition you already have in the catalog for that package.

Reply:
Perhaps `manages` instead of `installs` is better? Also, due to the behavior of `ensure_resource(s)` the state must match. So perhaps `ensures` is even better.
```suggestion
# @summary Takes a list of packages and ensures they exist
```